### ISSUE TITLE
NaN issue running LINE

### ISSUE THREAD

**Issue by xinyangz** [thunlp/OpenNE#21]
I have encountered NaN in the loss as well as the learned embedding when running with LINE. Same issue does not occur when running with the [code](https://github.com/tangjianpku/LINE) provided by the LINE paper author.

The command I used was:

`python src/main.py --input graph_memes.txt --graph-format edgelist --representation-size 64 --directed --epochs 20 --workers 32 --method line --output memes_line.txt`

The command line output was:

> Reading...
Pre-procesing for non-uniform negative sampling!
Instructions for updating:
Use the retry module or similar alternatives.
Pre-procesing for non-uniform negative sampling!
epoch:0 sum of loss:9389.645792722702
epoch:0 sum of loss:4735.700150832534
epoch:1 sum of loss:9331.65645313263
epoch:1 sum of loss:4152.164891295135
epoch:2 sum of loss:9321.187238454819
epoch:2 sum of loss:3999.570925347507
epoch:3 sum of loss:9314.46668368578
epoch:3 sum of loss:3914.669999137521
epoch:4 sum of loss:9310.313271224499
epoch:4 sum of loss:3865.22195328027
epoch:5 sum of loss:9308.589637458324
epoch:5 sum of loss:3833.3185774832964
epoch:6 sum of loss:9306.87627118826
epoch:6 sum of loss:3810.1825504228473
epoch:7 sum of loss:9304.062089920044
epoch:7 sum of loss:3792.079162888229
epoch:8 sum of loss:9303.782062470913
epoch:8 sum of loss:3780.837639503181
epoch:9 sum of loss:9303.198790431023
epoch:9 sum of loss:3770.1571313217282
epoch:10 sum of loss:9302.51797068119
epoch:10 sum of loss:3763.195652872324
epoch:11 sum of loss:9301.631554305553
epoch:11 sum of loss:3758.490410581231
epoch:12 sum of loss:9300.383490085602
epoch:12 sum of loss:3752.485915541649
epoch:13 sum of loss:9300.618901848793
epoch:13 sum of loss:3749.2992250844836
epoch:14 sum of loss:9299.43757379055
epoch:14 sum of loss:3747.6589295864105
epoch:15 sum of loss:9299.78179126978
epoch:15 sum of loss:3743.776451356709
epoch:16 sum of loss:9300.20017015934
epoch:16 sum of loss:3739.654469586909
epoch:17 sum of loss:nan
epoch:17 sum of loss:3739.8969665542245
epoch:18 sum of loss:nan
epoch:18 sum of loss:3736.818851336837
epoch:19 sum of loss:nan
epoch:19 sum of loss:3736.8378988951445
3631.65814495
Saving embeddings...

The input file is attached.

[graph_memes.txt.zip](https://github.com/thunlp/OpenNE/files/1866977/graph_memes.txt.zip)

**Comment by zzy14** [comment]
Hi,

I use `tf.log_sigmoid()` instead of `tf.log(tf.nn.sigmoid())` to solve this problem.

Please pull the new version.
